source/common/http/conn_manager_impl.cc

```diff
@@ -260,6 +260,14 @@ void ConnectionManagerImpl::doDeferredStreamDestroy(ActiveStream& stream) {
 
   read_callbacks_->connection().dispatcher().deferredDelete(stream.removeFromList(streams_));
 
+  // The response_encoder should never be dangling as the codec level
+  // stream will either outlive the ActiveStream, or be alive in deferred
+  // deletion queue at this point.
+  if (stream.response_encoder_) {
+    stream.response_encoder_->getStream().removeCallbacks(stream);
+    stream.response_encoder_ = nullptr;
```

Comment:
If we're going to null this out, I'm inclined to think we should have nullptr error handling places where we reference it (though feel free to wait on second opinion of Matt's when he's back)

Reply:
We know from `newStream()` that this value is set, and given the flow of the active stream lifecycle it should be valid in all other locations as is. Nothing else should really be touching this stream after this point apart from maybe crash dumping -- in which case having it touch a dangling pointer if not nulled out would crash the crash dumping 
 :( .

What would be the right way to recover from this being nullptr in places where it's currently used? It seems like it's an invariant in those places that it is set.

Reply:
but the dumping code can be used in logs as well, so in the medium term we could end up crashing folks with debug logging turned on.  If we can't handle it being null maybe we shouldn't set it to null?

Reply:
🤦  Will revert setting this to nullptr -- with the refactor this won't be dangling to nothing just to something that might already be in the deferred delete queue. Thanks for pointing this out 🤦‍♂️ 🤦 🤦‍♀️